src/components/Home/WeekRate.tsx

```diff
@@ -0,0 +1,144 @@
+import { Cell, Label, Pie, PieChart } from 'recharts';
+import styled from 'styled-components';
+import { WeekRateProps } from '../../types/Week/weekRateTypes';
+
+const CustomLabel = ({ viewBox, value }: WeekRateProps) => {
+  const { cx, cy } = viewBox;
+
+  return (
+    <>
+      <StyledText
+        x={cx}
+        y={cy - 20}
+        dominantBaseline="middle"
+        textAnchor="middle"
+      >
+        {value}
+      </StyledText>
+    </>
+  );
+};
+const WeekRate = () => {
+  const maxProblems = 0;
+  const solvedProblems = 0;
+
+  const percentage =
+    maxProblems && solvedProblems ? (solvedProblems / maxProblems) * 100 : 10;
+  const chartData = [{ name: 'Solved', value: percentage }];
+  const endAngle = 180 - (percentage / 100) * 180;
+  return (
+    <Container>
+      <Header>주간 성과율</Header>
+      <ChartContainer>
+        <PieChart width={300} height={140}>
+          <defs>
+            <linearGradient id="linear" x1="0" y1="0" x2="1" y2="1">
+              <stop offset="100%" stopColor="#CA73FF" />
+              <stop offset="100%" stopColor="#000000" />
+            </linearGradient>
+          </defs>
+          <Pie
+            data={chartData}
+            dataKey="value"
+            startAngle={180}
+            endAngle={0}
+            innerRadius={95}
+            outerRadius={115}
+            stroke="none"
+            cornerRadius={10}
+            paddingAngle={0}
+            cy={126}
+            cx="50%"
+            fill="#34363C"
+          >
+            <Label
+              position="center"
+              content={
+                <CustomLabel
+                  value={
+                    maxProblems && solvedProblems
+                      ? `${percentage.toFixed(0)}%`
+                      : `${0}%`
+                  }
+                  viewBox={{ cx: 0, cy: 100 }}
+                />
+              }
+            />
+          </Pie>
+          <Pie
+            data={chartData}
+            dataKey="value"
+            startAngle={180}
+            endAngle={endAngle}
+            innerRadius={95}
+            outerRadius={115}
+            stroke="none"
+            cornerRadius={10}
+            paddingAngle={0}
+            cy={126}
+            cx="50%"
+          >
+            <Cell key="cell-1" fill={'url(#linear)'} />
+          </Pie>
+        </PieChart>
+      </ChartContainer>
+      <MessageContainer>
+        {maxProblems && solvedProblems ? (
+          <>
+            <Message>축하해요!</Message>
+            <Message>지난 주보다 {solvedProblems}문제 더 풀었어요!</Message>
+          </>
+        ) : (
+          <>
+            <Message>이번 주에 푼 문제가 없어요</Message>
+            <Message>문제 풀이 인증하고 그래프를 채워보세요</Message>
+          </>
+        )}
```

Comment:
P2: 이렇게 하면 코드가 더 깔끔해질 것 같아요! 이름은 대충 지은거라 좀 더 ㄱ직관적인 이름으로 해주세요 !

```suggestion
          const topMsg = isProblemsExit ? "축하해요" : "이번 주에 푼 문제가 없어요";
          const bottomMsg = isProblemsExit ? "지난 주보다 {solvedProblems}문제 더 풀었어요!" : "문제 풀이 인증하고 그래프를 채워보세요";
          
          <>
            <Message>{ topMsg }</Message>
            <Message>{ bottomMsg }</Message>
          </>
```